src/cmd/bench.rs

```diff
@@ -13,19 +13,26 @@ pub struct Cmd {
     /// Number of iterations to use for test
     #[arg(long, short, default_value_t = 100)]
     pub iterations: u32,
+    /// Flag to enable durations in the results
+    #[arg(long, short)]
+    pub durations: bool,
 }
 
 impl Cmd {
     pub fn run(&self, device: &Device) -> Result {
         let keypair = device.get_keypair(false)?;
-        let duration = bench_sign(&keypair, self.iterations)?;
+        let (duration, durations) = bench_sign(&keypair, self.iterations)?;
         let rate = self.iterations as f64 / duration.as_secs_f64();
         let avg_ms = duration.as_millis() as f64 / self.iterations as f64;
-        let json = json!({
+        let mut json = json!({
             "iterations": self.iterations,
             "avg_ms": round2(avg_ms),
             "rate": round2(rate),
         });
+        if self.durations {
```

Comment:
Why not map durations to a Vec of mills and then just insert it into the top level json?
e.g.

```rust
let durations: Vec<u128> = durations.map(|d| d.as_millis()).collect();
```

and in the json construction above:

```rust 
let json = json!({
    "iterations": self.iterations,
    "avg_ms": round2(avg_ms),
    "rate": round2(rate),
    "durations": durations,
 });
```